### PR TITLE
Fix `convert_hf_to_dcp`

### PR DESCRIPTION
**Pull request by CarlOwOs**
### Summary

- `convert_hf_to_dcp` previously saved weights under a nested `{"model": state_dict}` key, producing `model.model.*` entries.
- `flame/torchtitan` expects a flat `model.*` state dict for seed checkpoints; the nested layout broke `dcp.load` when resuming training.
- Updated the converter to save the flat state dict directly (`model.*` keys), matching the loader’s expectations.

### Testing

- Converted Qwen3-1.7B from Hugging Face using `python -m flame.utils.convert_hf_to_dcp --model Qwen/Qwen3-1.7B --checkpoint <path>/checkpoint/step-0` and verified keys are `model.*` (no `model.model.*`).
- Ran `train.sh` for continual training with that checkpoint; training now proceeds past checkpoint load.

## Summary by CodeRabbit

* **Changes**
  * Updated the model weight storage format when converting from HuggingFace models. The saved checkpoint structure has been simplified, which may affect dependent workflows that rely on the previous model structure.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by CarlOwOs** [comment]
Need to change this line in `convert_dcp_to_hf`:
- model.load_state_dict(torch.load(checkpoint_path, map_location='cpu')['model'])

I also need to check if the model key is used elsewhere throughout the code.
